respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -191,6 +188,29 @@
 
 
     /**
+     * A listener interface to receive a notification when the request to retrieve the history
+     * of messages for a list of groups has completed
+     */
+    public interface GroupHistoriesCompletionListener {
+
+        void onSuccess(HashMap groupsToMessages);
+
+        void onError(String errorMessage);
+    }
+
+    /**
+     * A listener interface to receive a notification when the request to retrieve the
+     * history of messages for a specific group has completed
+     */
+    public interface GroupHistoryCompletionListener {
+
+        void onSuccess(ArrayList<RespokeGroupMessage> messageList);
```

Comment:
I would specify List here instead of ArrayList. The implementation detail of the map shouldn't be part of the API.